lib/processors/webgl2/pipelines/fastBilateralFilterStage.ts

```diff
@@ -18,6 +18,11 @@ export function buildJointBilateralFilterStage(
   outputTexture: WebGLTexture,
   canvas: HTMLCanvasElement
 ) {
+  // NOTE(mmalavalli): This is a faster approximation of the joint bilateral filter.
+  // It improves the complexity from O(w x h x r^2) to O(w x h x r), where:
+  // w => width of the output video frame
+  // h => height of the output video frame
+  // r => radius of the joint bilateral filter kernel
```

Comment:
Nice. Is this something you wrote or do you have a reference somewhere that provides more details about this algorithm?